src/lib/pages/account-details/data.ts

```diff
@@ -152,3 +201,256 @@ export const useUserAssetInfos = (
+    totalDelegations: undefined,
+    delegations: undefined,
+    isLoadingDelegations: isLoading || isLoadingRawDelegations,
+    totalUnbondings: undefined,
+    unbondings: undefined,
+    isLoadingUnbondings: isLoading || isLoadingRawUnbondings,
+    totalRewards: undefined,
+    rewards: undefined,
+    isLoadingRewards: isLoading || isLoadingRawRewards,
+    redelegations: undefined,
+    isLoadingRedelegations: isLoading || isLoadingRawRedelegations,
+    totalCommission: undefined,
+    isLoadingTotalCommission: isLoading || isLoadingRawCommission,
+  };
+
+  if (rawStakingParams && assetInfos && validators) {
+    const stakingParams = {
+      ...rawStakingParams,
+      ...assetInfos[rawStakingParams.bondDenom],
+    };
+    data.stakingParams = stakingParams;
+
+    data.isValidator = Object.keys(validators).includes(
+      addrToValoper(walletAddress)
+    );
+
+    data.delegations = rawDelegations?.map<Delegation>((raw) => ({
+      validator: {
+        validatorAddress: raw.validatorAddress,
+        moniker:
+          validators[raw.validatorAddress]?.moniker ?? raw.validatorAddress,
+      },
+      token: coinToTokenWithValue(raw.denom, raw.amount, assetInfos[raw.denom]),
+    }));
+    data.totalDelegations = data.delegations?.reduce(
+      (total, delegation) => ({
+        ...total,
+        [delegation.token.denom]: addTotal(
+          total[delegation.token.denom],
+          delegation.token
+        ),
+      }),
+      {} as Record<string, TokenWithValue>
+    );
+
+    data.unbondings = rawUnbondings?.map<Unbonding>((raw) => ({
+      validator: {
+        validatorAddress: raw.validatorAddress,
+        moniker:
+          validators[raw.validatorAddress]?.moniker ?? raw.validatorAddress,
+      },
+      completionTime: raw.completionTime,
+      token: coinToTokenWithValue(
+        rawStakingParams.bondDenom,
+        raw.amount,
+        assetInfos[rawStakingParams.bondDenom]
+      ),
+    }));
+    data.totalUnbondings = data.unbondings?.reduce(
+      (total, unbonding) => ({
+        ...total,
+        [unbonding.token.denom]: addTotal(
+          total[unbonding.token.denom],
+          unbonding.token
+        ),
+      }),
+      {} as Record<string, TokenWithValue>
+    );
+
+    data.rewards = rawRewards?.rewards.reduce(
+      (prev, raw) => ({
+        ...prev,
+        [raw.validatorAddress]: raw.reward.map<TokenWithValue>((coin) =>
+          coinToTokenWithValue(coin.denom, coin.amount, assetInfos[coin.denom])
+        ),
+      }),
+      {} as Record<string, TokenWithValue[]>
+    );
+    data.totalRewards = rawRewards?.total.reduce(
+      (total, raw) => ({
+        ...total,
+        [raw.denom]: coinToTokenWithValue(
+          raw.denom,
+          raw.amount,
+          assetInfos[raw.denom]
+        ),
+      }),
+      {} as Record<string, TokenWithValue>
+    );
+
+    data.redelegations = rawRedelegations?.map<Redelegation>((raw) => ({
+      ...raw,
```

Comment:
Do we need this? Since `redelegations` is type `Option<Redelegation[]>`

Reply:
this change from type `Option<RawRedelegation[]>` to `Option<Redelegation[]>`

Reply:
Remove `...raw`